### ISSUE TITLE
Numeric keyboard does not appear for numeric input unless a regular input was already selected

### ISSUE THREAD

**Issue by TheGreenAirplane** [canonical/ubuntu-frame-osk#76]
I have an input element, <input type="number">, which is supposed to trigger the numpad layout. However, it only shows up if I've had another input already selected. If I select a non-numeric input, the regular keyboard comes up, if I then select the numeric input, I get the numpad. 

But if I select the numeric input first, I get the regular keyboard. If I deselect an input, causing the keyboard to hide, and then select a numeric input, I get the regular keyboard again. 

The numeric keyboard should show up consistently, every time a numeric input is selected.

I'm running the latest (as of 5 days ago) versions of Ubuntu Server, Ubuntu Frame, Mir Kiosk, and Ubuntu Frame OSK.

**Comment by Saviq** [comment]
Hi @TheGreenAirplane, could you please try the `24/edge` channel of `ubuntu-frame-osk`?

It has a much newer version of the OSK software itself ([squeekboard](https://gitlab.gnome.org/World/Phosh/squeekboard)).

**Comment by TheGreenAirplane** [comment]
Hi, thank you for your response. I installed the keyboard via snap:
sudo snap install ubuntu-frame-osk

if you could elaborate what exactly is this 24/edge channel, I'll be sure to give it a try.

**Comment by TheGreenAirplane** [comment]
Ok so I asked ChatGpt what is a "24/edge channel" and with its help I was able to install ubuntu-frame-osk from the 24/edge channel. My changes to the layout are still being ignored, but at least now the keyboard is 3 times smaller than it should be. So it's definitely progress, just not in the right direction :)

**Comment by Saviq** [comment]
Yeah, the being smaller is why we've not updated to the newer version (#46)… Thanks for confirming.

I could confirm your problem using url=https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_number. I suspect it's an upstream issue, will try and confirm and report it there.

[Nagranie ekranu z 2024-06-18 07-00-35.webm](https://github.com/canonical/ubuntu-frame-osk/assets/231077/a8cd703a-cd14-4d70-86e7-f16f29be5958)

**Comment by TheGreenAirplane** [comment]
Hi, is there any update on this? It's causing us issues in production. Apparently if you ask people to type in a number, and don't give them a purely numeric keyboard, they get confused and type in the letter "o" instead of zero. 
Has this been escalated to Squeekboard devs? Or should I do it? Did you manage to confirm that the problem is with Squeekboard code and not ubuntu-frame-osk?

**Comment by Saviq** [comment]
@TheGreenAirplane there are three components here that need to work in unison:
- Squeekboard (ubuntu-frame-osk is just a distribution of it)
- a compositor (Mir, in the case of Ubuntu Frame)
- client application ([WPE kiosk snap](https://snapcraft.io/wpe-webkit-mir-kiosk) here)

It would be great if you could try reproducing the problem replacing these components (e.g., using Sway as the compositor, or building upstream Squeekboard). That could help pinpoint where the problem lies.

I'm not aware of a different keyboard software implementing the same [Wayland protocols](https://gitlab.gnome.org/World/Phosh/squeekboard/-/tree/main/protocols?ref_type=heads), but browsing through GitHub might yield one:

https://github.com/search?q=virtual-keyboard-unstable-v1.xml&type=code

**Comment by TheGreenAirplane** [comment]
Hello, I've been told to inquire about the status of this bug. Has there by any chance been some progress on this?

**Comment by AlanGriffiths** [comment]
@TheGreenAirplane it would helpful if [you could do some investigation](https://github.com/canonical/ubuntu-frame-osk/issues/76#issuecomment-2228030109) to clarify which component is misbehaving.

**Comment by TheGreenAirplane** [comment]
I am not an advanced Linux user, I don't have the skill set to perform these tests.

**Comment by Saviq** [comment]
@TheGreenAirplane sorry, it will have to wait until we have some free time on our hands. Because of all the layers ultimately resulting in this issue, it's not at the top of our backlog.

**Comment by tarek-y-ismail** [comment]
Possible related to https://github.com/canonical/mir/issues/3580?
